src/Infrastructure.VS/LanguageDetection/SonarLanguageRecognizer.cs

```diff
@@ -154,6 +159,16 @@ private string GetNormalizedExtention(string fileName)
             return extension.ToLowerInvariant();
         }
         
-
+        private bool IsFileNameInvalid(string fileName)
```

Comment:
https://stackoverflow.com/questions/27253394/getinvalidfilenamechars-doesnt-contain-all-illegal-chars

I think it's better to do a try-catch around GetNormalizedExtention

Reply:
I don't think it's a concern for us. Our code will always run on a windows machine. This would create problems if we had operations run on different systems.